backend/backend/models/tables.py

```diff
@@ -48,21 +50,35 @@ class Order(SQLModel, table=True):
     providers: list["Provider"] = Relationship(back_populates="orders", link_model=OrderProviderLink)
 
 
+### User
 class UserType(str, enum.Enum):
     MP_USER = "mp_user"
     PROVIDER_MANAGER = "provider_manager"
     COORDINATOR = "coordinator"
     ADMIN = "admin"
 
 
-class User(SQLModel, table=True):
-    id: int | None = Field(default=None, primary_key=True)
+class UserBase(SQLModel):
     name: str = Field(nullable=False, min_length=1)
-    created_at: datetime = Field(default_factory=lambda: datetime.now(tz=timezone.utc))
-    updated_at: datetime = Field(default_factory=lambda: datetime.now(tz=timezone.utc))
     email: str = Field(nullable=False, unique=True, regex=r"^\S+@\S+\.\S+$")
     user_type: list[UserType] = Field(sa_column=Column(ARRAY(Enum(UserType)), nullable=False), min_length=1)
 
+
+class UserPublic(UserBase):
+    id: int
+    created_at: datetime
+    updated_at: datetime
+
+
+class UserCreate(UserBase):
+    pass
+
+
+class User(UserBase, table=True):
```

Comment:
We need to decide which properties should be kept in keycloak, and what needs to be also present in BOS' DB. 

I think that we should keep a detailed description of a user in keycloak and only a minimal subset in the service db (service specific with a reference to keycloak's user who can be fully fetched on demand).

In service db:
- internal id
- Keycloak user_id
- last login
- role snapshot
- provider
- business-specific user settings for a given service (preferences, ... whatever) 

That is a loose proposition, opened to discussion. Then we need to use Keycloak webhooks or event listeners to update BOS when users change in keycloak.